python/tank/bootstrap/resolver.py

```diff
@@ -166,11 +129,43 @@ def resolve_configuration(
         # create an object to represent our configuration install
         return Configuration(
             config_root,
-            self._sg_connection,
+            sg_connection,
             cfg_descriptor,
-            project_id,
+            self._project_id,
             None,  # pipeline config id
             self._bundle_cache_fallback_paths
         )
 
+    def resolve_shotgun_configuration(
+        self,
+        pipeline_config_name,
+        fallback_config_descriptor,
+        sg_connection
+    ):
+        """
+        Return a configuration object by requesting a pipeline configuration
+        in Shotgun. If no suitable configuration is found, return a configuration
+        for the given fallback config.
+
+        :param pipeline_config_name: Name of configuration branch (e.g Primary)
+        :param fallback_config_descriptor: descriptor dict or string for fallback config.
+        :param sg_connection: Shotgun API instance
+        :return: :class:`Configuration` instance
+        """
+        log.debug(
+            "%s resolving configuration from Shotgun Pipeline Configuration %s" % (self, pipeline_config_name)
+        )
+
+        log.warning(
+            "Shotgun pipeline configuration resolve has not been implemented yet. "
+            "Falling back on the base configuration resolver."
+        )
```

Comment:
The warning is intentional so that you can begin developing a plugin with the *intention* to make it overridable in the future. Right now, core doesn't support it - but hopefully will very soon! The current behaviour is a graceful fallback on the base configuration, e.g. exactly the case that you would get if you don't have any pipeline config entry for it in Shotgun. So I think this is a smoother way to transition than to rise. 

This way, the interface is set right out of the gate and once we add the business logic for the shotgun overrides, all plugins become shotgun configurable without the plugins themselves needing to change. Another reason is also backwards compatibility with older versions of dev plugins, to make sure things won't explode (these plugins won't have the  `do_shotgun_config_lookup` property set to `False` yet) . Hope it makes sense :)